File: Pset6/sentimental-readability/readability.py
# Using CS50 library's getstring function only
from cs50 import get_string

# defining the main function and introducing text, letters, words and sentences as variables
def main():
    text = get_string("Text: ")
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    # Computing Coleman-Liau index
    L = (letters / words) * 100
    S = (sentences / words) * 100
    index = round(0.0588 * L - 0.296 * S - 15.8)

    # Outputting grade level based on index
    if index > 0 and index <= 16:
        print(f"Grade: {index}")
    elif index > 16:
        print("Grade 16+")
    else:
        print("Before Grade 1")


# function to calculate letters
def count_letters(text):
    count = 0
    for char in text:
        if char.isalpha():
            count += 1

    return count


# function to calculate words
def count_words(text):
    count = 1
    i = 1
    length = len(text)
    while i < length:
        if text[i] == " " and text[i - 1] != " ":
            count += 1
        i += 1

    return count


# function to calculate sentences
def count_sentences(text):
    count = 0
    for char in text:
        if char in [".", "?", "!"]:
            count += 1

    return count


main()

File: Pset2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

// initiate abstraction for letters, words and sentences;
int count_letters(string text);
int count_words(string text);
int count_sentences(string text);
int main(void)
{
    // get input from user
    string text = get_string("Type in your sentence:  ");

    // count the number of letters, words and sentences in a text

    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);
    // Use the formula and print answer
    float index = 0.0588 * ((float) letters * 100 / (float) words) - 0.296 * ((float) sentences * 100 / (float) words) - 15.8;
    int indexLH = round(index);
    
    if (indexLH < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (indexLH > 16)

    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", indexLH);
    }
}
// Abstraction for letters
// Return the number of letters
int count_letters(string text)

{
    int count_letters = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))

        {
            count_letters++;
        }
    }

    return count_letters;
}

// Abstraction for words
// Return the number of words
int count_words(string text)
{
    int count_words = 1;
    for (int j = 0; j < strlen(text); j++)
    {
        if (text[j] == ' ')

        {
            count_words++;
        }
    }
    return count_words;
}
// Abstraction for sentences
// Return the number of sentences
int count_sentences(string text)

{
    int count_sentences = 0;
    int n = strlen(text);
    for (int k = 0; k < n; k++)
    {
        if (text[k] == '.' || text[k] == '?' || text[k] == '!')

        {
            count_sentences++;
        }
    }
    return count_sentences;
}